cli/lib/kontena/cli/cloud/login_command.rb

```diff
@@ -137,7 +137,7 @@ def web_flow
       puts
 
       server_thread  = Thread.new { Thread.main['response'] = web_server.serve_one }
-      browser_thread = Thread.new { Launchy.open(uri.to_s) }
+      browser_thread = Thread.new { Kontena::Cli::BrowserLauncher.open(uri.to_s) }
```

Comment:
> The launch is happening inside a thread and the raise will be eaten.

Using `Thread.new { ... }` like this without any explicit rescue isn't necessarily that great, but that hasn't changed in this PR?

Reply:
It's not 100% clear to me how `thread.join` behaves, but at least the example in the docs for [`thread.value`](https://ruby-doc.org/core-2.3.1/Thread.html#method-i-value) shows it being used to handle any top-level exception raised in the thread:

```ruby
b = Thread.new { raise 'something went wrong' }
b.value   #=> RuntimeError: something went wrong
```

Reply:
I don't think the browser thread is needed for anything. In case the detection fails, the error raised will not be seen and it just hangs forever waiting.